client/components/data/query-jetpack-plugins/index.jsx

```diff
@@ -1,48 +1,6 @@
-import { isEqual } from 'lodash';
-import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
-import { fetchPlugins } from 'calypso/state/plugins/installed/actions';
-import { isRequestingForSites } from 'calypso/state/plugins/installed/selectors';
+import usePluginsQuery from 'calypso/data/plugins/installed/use-plugins-query';
 
-class QueryJetpackPlugins extends Component {
-	static propTypes = {
-		siteIds: PropTypes.arrayOf(
-			PropTypes.oneOfType( [ PropTypes.string, PropTypes.number ] ).isRequired
-		).isRequired,
-		isRequestingForSites: PropTypes.bool,
-		fetchPlugins: PropTypes.func,
-	};
-
-	componentDidMount() {
-		if ( this.props.siteIds && ! this.props.isRequestingForSites ) {
-			this.props.fetchPlugins( this.props.siteIds );
-		}
-	}
-
-	componentDidUpdate( prevProps ) {
-		if ( isEqual( prevProps.siteIds, this.props.siteIds ) ) {
-			return;
-		}
-		this.refresh( this.props.isRequestingForSites, this.props.siteIds );
-	}
-
-	refresh( isRequesting, siteIds ) {
-		if ( ! isRequesting ) {
-			this.props.fetchPlugins( siteIds );
-		}
-	}
-
-	render() {
-		return null;
-	}
+export default function QueryJetpackPlugins( props ) {
+	usePluginsQuery( props.siteIds );
```

Comment:
Good call!

Reply:
I looked into doing this but it turns out that the task is quite involved there are quite a few places where the QueryJetpackPlugins is used. I would rather save this fast for another follow up PR which would hopefully make it so that the changes are easier to test.

Reply:
Sounds good, but in the meantime can we use `usePluginsQuery` separately then? Because as you said, `QueryJetpackPlugins` is used in quite a few places and if we alter it in this PR, we have to test all of those places.